### PR TITLE
remove bogus data

### PR DESCRIPTION
**Pull request by aradmargalit**
Filters out impossible data points

### CODE REVIEW

File: src/apiClients/stravaClient/converters.ts [hunk 1]
```diff
@@ -48,7 +48,10 @@ export function convertStravaActivityResponse({
     const { time, latlng, altitude } = streamsResponse;
     const startTime = DateTime.fromISO(response.start_date);
 
-    const times = time.data.map((t) => startTime.plus({ seconds: t }).toISO());
+    const times = time.data
+      .map((t) => startTime.plus({ seconds: t }).toISO())
+      .filter((x): x is string => Boolean(x));
```
Comment (aradmargalit) [inline src/apiClients/stravaClient/converters.ts:53]: only to appease TS

File: src/utils/garminUtils.ts [hunk 2]
```diff
@@ -36,17 +75,15 @@ export async function garminActivityFromFile(
   const messages = decoder.read().messages;
   const garminRecords: GarminActivityRecord[] = messages.recordMesgs;
 
-  // TODO: update these
-  // @ts-ignore
```
Comment (aradmargalit) [inline src/utils/garminUtils.ts:40]: This worked as-is

File: src/utils/garminUtils.ts [hunk 3]
```diff
@@ -27,6 +29,37 @@ function convertGarminRecord(garminRecord: GarminActivityRecord): Record {
...
+    return true;
+  }
+
+  // Assume the first coordinate is good
+  const goodCoordinate = records[0].coord;
+  const currentCoordinates = record.coord;
+  const distance = lawOfCosinesDistance(goodCoordinate, currentCoordinates);
+
+  return distance <= IMPOSSIBLE_INCREMENTAL_DISTANCE_METERS;
+}
```
Comment (eshedmargalit) [inline src/utils/garminUtils.ts:61]: Doesn't this function only check if the distance between the first point and the `index`th point is below the threshold? There's no comparison to the previous point as far as I can tell; it always compares to the first point. That probably also works as a filtering method, unless the activity is a month-long spiritual journey

File: src/utils/garminUtils.test.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+describe('garminUtils', () => {
+  describe(isDistancePossible.name, () => {
+    it('returns true for the first record', () => {
+      expect(isDistancePossible(makeMockRecord(), 0, [])).toBe(true);
+    });
+
+    it('returns true when records are reasonably close', () => {
+      const first: Coordinate = [37.709008223518005, -121.92952352047195]; // Chuck-e-cheese Dublin CA
+      const second: Coordinate = [37.706358213369164, -121.92775663545241]; // Super Cuts Dublin CA
```
Comment (eshedmargalit) [inline src/utils/garminUtils.test.ts:19]: lol

**Review by eshedmargalit — APPROVED** [review]

File: src/utils/garminUtils.ts (hunk 3 above)
Comment (aradmargalit) [reply src/utils/garminUtils.ts]: Yeah good catch, I had changed the implementation but forgot the comments. Updated

**Pull request merged**